src/query/api/v1/handler/prometheus/remote/read_test.go

```diff
@@ -204,7 +204,11 @@ func TestPromReadStorageWithFetchError(t *testing.T) {
 	}
 
 	fetchOpts := &storage.FetchOptions{}
-	result := storage.NewPromResult([]*prompb.TimeSeries{})
```

Comment:
I'm reverting this small refactor because it was ill motivated. The caller should init the inner `Metadata` if it needs to.